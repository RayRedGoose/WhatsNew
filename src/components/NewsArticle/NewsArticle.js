import React from 'react';
import './NewsArticle.scss';
import ArrowIcon from '../../images/arrow.svg';
import PropTypes from 'prop-types';

const NewsArticle = ({ newsPreviewImage, title, intro, articleLink }) => {
  return (
    <section className="news-preview">
      <header style={{backgroundImage: `url(${newsPreviewImage})`}}>
      </header>
      <div className="text-wrapper">
        <h2>{title}</h2>
        <p>{intro}</p>
      </div>
      <div className="link-wrapper">
        <a href={articleLink}>Link To Article</a>
        <img src={ArrowIcon} alt="arrow forward"/>
      </div>
    </section>
  );
}

NewsArticle.propTypes = {
  newsPreviewImage: PropTypes.string,
  title: PropTypes.string,
  intro: PropTypes.string,
  articleLink: PropTypes.string
}

export default NewsArticle;
