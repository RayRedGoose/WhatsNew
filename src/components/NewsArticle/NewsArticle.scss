.news-preview {
  background-color: #fff;
  overflow: hidden;
  text-align: center;
  width: 49%;
  min-width: 300px;
  margin-bottom: 20px;
  position: relative;
  header {
    height: 180px;
    overflow: hidden;
    background-position: center;
    background-size: 100%;
    background-repeat: no-repeat;
  }
  .text-wrapper {
    height: auto;
    margin-bottom: 60px;
    h2 {
      padding: 20px;
    }
    p {
      padding: 0 20px;
    }
  }
  .link-wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: #b66af0;
    width: 100%; height: 40px;
    position: absolute;
    bottom: 0;
    a {
      color: #fff;
      font-size: 24px;
      text-decoration: none;
    }
    img {
      width: 30px;
      margin-left: 40px;
    }
  }
}

@media only screen and (max-width: 910px) {
  .news-preview {
    width: 100%;
  }
}
