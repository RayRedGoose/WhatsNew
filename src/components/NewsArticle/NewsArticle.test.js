import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

it('should match the snapshot with all data passed in correctly', () => {
  const element = shallow(<NewsArticle
    newsPreviewImage="./image-preview.jpg"
    title="Article"
    intro="description of article"
    articleLink="link.com"
    key="1"
  />);

  expect(element).toMatchSnapshot();
});
