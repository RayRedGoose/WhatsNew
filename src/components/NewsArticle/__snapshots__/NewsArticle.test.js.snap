// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot with all data passed in correctly 1`] = `
<section
  className="news-preview"
>
  <header
    style={
      Object {
        "backgroundImage": "url(./image-preview.jpg)",
      }
    }
  />
  <div
    className="text-wrapper"
  >
    <h2>
      Article
    </h2>
    <p>
      description of article
    </p>
  </div>
  <div
    className="link-wrapper"
  >
    <a
      href="link.com"
    >
      Link To Article
    </a>
    <img
      alt="arrow forward"
      src="arrow.svg"
    />
  </div>
</section>
`;
